benchmarks/src/main/java/org/questdb/Decimal256SubtractBenchmark.java

```diff
@@ -0,0 +1,189 @@
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+
+import java.math.BigDecimal;
+import java.math.MathContext;
+import java.math.RoundingMode;
+import java.util.concurrent.TimeUnit;
+
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+@Warmup(iterations = 5, time = 100, timeUnit = TimeUnit.MILLISECONDS)
+@Measurement(iterations = 5, time = 100, timeUnit = TimeUnit.MILLISECONDS)
+@Fork(1)
+@State(Scope.Benchmark)
+public class Decimal256SubtractBenchmark {
+
+    private BigDecimal bigDecimalMinuend;
+    private BigDecimal bigDecimalSubtrahend;
+    private Decimal256 decimal256Minuend;
+    private Decimal256 decimal256Subtrahend;
+    private Decimal256 decimal256Result;
+    private MathContext mathContext;
+    @Param({"SIMPLE", "LARGE_NUMBERS", "SMALL_NUMBERS", "DIFFERENT_SCALES", "NEAR_ZERO", "NEGATIVE_SUBTRACTION"})
+    private String scenario;
+
+    public static void main(String[] args) throws RunnerException {
+        Options opt = new OptionsBuilder()
+                .include(Decimal256SubtractBenchmark.class.getSimpleName())
+                .warmupIterations(5)
+                .measurementIterations(10)
+                .forks(1)
+                .build();
+
+        new Runner(opt).run();
+    }
+
+    @Benchmark
+    public BigDecimal bigDecimalSubtract() {
+        return bigDecimalMinuend.subtract(bigDecimalSubtrahend);
+    }
+
+    @Benchmark
+    public BigDecimal bigDecimalSubtractWithContext() {
+        return bigDecimalMinuend.subtract(bigDecimalSubtrahend, mathContext);
+    }
+
+    @Benchmark
+    public Decimal256 decimal256Subtract() {
+        decimal256Result.copyFrom(decimal256Minuend);
+        decimal256Result.subtract(decimal256Subtrahend);
+        return decimal256Result;
+    }
+
+    @Benchmark
+    public void decimal256Subtract256By128() {
+        // Test subtraction of 256-bit and 128-bit values
+        Decimal256 largeMinuend = new Decimal256();
+        largeMinuend.setFromLong(123456789098765432L, 6);
+
+        decimal256Result.copyFrom(largeMinuend);
+        decimal256Result.subtract(decimal256Subtrahend);
+    }
+
+    @Benchmark
+    public void decimal256Subtract128Bit() {
+        // Test subtraction of two 128-bit values
+        Decimal256 val1 = Decimal256.fromDouble(987654.321, 3);
+        Decimal256 val2 = Decimal256.fromDouble(123456.789, 3);
+
+        decimal256Result.copyFrom(val1);
+        decimal256Result.subtract(val2);
+    }
+
+    @Benchmark
+    public void decimal256SubtractNearZero() {
+        // Test subtraction that results in near-zero: 123.456 - 123.455
```

Comment:
bad test results are probably down to allocation inside the test, which outweighs the computation